core/src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -462,12 +467,13 @@ private void analyze(TokenStream stream, Analyzer analyzer, String field, Set<St
         return extendedAttributes;
     }
 
-    private static CharFilterFactory[] getCharFilterFactories(AnalyzeRequest request, IndexSettings indexSettings, AnalysisRegistry analysisRegistry,
-                                                              Environment environment, CharFilterFactory[] charFilterFactories) throws IOException {
+    private static void parseCharFilterFactories(AnalyzeRequest request, IndexSettings indexSettings, AnalysisRegistry analysisRegistry,
+                                                              Environment environment, List<CharFilterFactory> charFilterFactoryList) throws IOException {
         if (request.charFilters() != null && request.charFilters().size() > 0) {
-            charFilterFactories = new CharFilterFactory[request.charFilters().size()];
-            for (int i = 0; i < request.charFilters().size(); i++) {
-                final AnalyzeRequest.NameOrDefinition charFilter = request.charFilters().get(i);
+            List<AnalyzeRequest.NameOrDefinition> charFilters = request.charFilters();
+            for (int i = 0; i < charFilters.size(); i++) {
+                AnalyzeRequest.NameOrDefinition charFilter = charFilters.get(i);
```

Comment:
let's do `for (AnalyzeRequest.NameOrDefinition charFilter : charFilters)` now that we do not need the index anymore